lib/active_merchant/billing/gateways/paysafe.rb

```diff
@@ -147,17 +145,20 @@ def add_address_for_vaulting(post, options)
 
       # This data is specific to creating a profile at the gateway's vault level
       def add_profile_data(post, payment, options)
-        address = options[:billing_address] || options[:address]
-
         post[:firstName] = payment.first_name
         post[:lastName] = payment.last_name
         post[:dateOfBirth] = {}
         post[:dateOfBirth][:year] = options[:date_of_birth][:year]
         post[:dateOfBirth][:month] = options[:date_of_birth][:month]
         post[:dateOfBirth][:day] = options[:date_of_birth][:day]
         post[:email] = options[:email] if options[:email]
-        post[:phone] = (address[:phone] || options[:phone]) if address[:phone] || options[:phone]
         post[:ip] = options[:ip] if options[:ip]
+
+        if options[:phone]
+          post[:phone] = options[:phone]
+        elsif address = options[:billing_address] || options[:address]
+          post[:phone] = address[:phone] if address[:phone]
+        end
```

Comment:
Just throwing this out there if you like the style 😃 

in `def add_profile_data`:
```
phone = phone(options)
post[:phone] = phone if phone
```
using the helper:
```
def phone(options)
  options[:phone] || options.dig(:billing_address, :phone) || options.dig(:address, :phone)
end
```